Fix log gathering

Deixx opened the pull request:
For DUTs without IP defined use host when creating log dir (IP set to None caused an exception)
Use paramiko method for copying log files (works when running tests from Windows)

pep8speaks commented:
Hello @mmichal10! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`test_package/stats/test_block_stats.py`](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py):

> [Line 31:12](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L31): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 32:12](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L32): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 34:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L34): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 36:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L36): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 37:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L37): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 39:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L39): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 40:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L40): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 42:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L42): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 43:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L43): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 48:5](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L48): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 49:5](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L49): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 78:80](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L78): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 84:21](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L84): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 87:81](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L87): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 91:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L91): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 92:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L92): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 95:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L95): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 100 characters)
> [Line 97:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L97): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 98:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L98): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 105:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L105): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 106:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L106): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 110:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L110): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 111:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L111): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 118:12](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L118): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 120:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L120): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 122:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L122): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 123:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L123): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 125:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L125): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 127:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L127): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 128:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L128): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 133:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L133): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 134:13](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L134): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 164:80](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L164): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 170:21](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L170): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 173:81](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L173): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 177:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L177): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 178:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L178): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 181:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L181): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 100 characters)
> [Line 183:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L183): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 184:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L184): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 191:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L191): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 192:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L192): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 196:29](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L196): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 197:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L197): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 202:1](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L202): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 228:101](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L228): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 238:48](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L238): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 238:50](https://github.com/Open-CAS/test-framework/blob/4232320c95a796af394a3f3c9ef185019bb2b4ca/test_package/stats/test_block_stats.py#L238): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals

katlapinka commented on test_utils/linux_command.py:
```diff
@@ -34,6 +34,11 @@ def remove_flag(self, flag):
         return self
 
     def set_param(self, key, *values):
+        try:
```
try/except can be cafely removed

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+
+import pytest
+import random
+import time
```
random, time not used

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+from test_package.test_properties import TestProperties
+from storage_devices.disk import DiskType
+from storage_devices.device import Device
+from test_tools.disk_utils import Filesystem, create_filesystem, mount, unmount
```
lines 19 & 20 not used

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+    (CacheMode.WA, write_wa_zero_stats), (CacheMode.WO, write_wo_zero_stats)]
+)
+@pytest.mark.parametrize(
+    "prepare_and_cleanup", [{"core_count": 3, "cache_count": 1}], indirect=True
```
"core_count": 3  - this means 3 disks to be used as cores, if you use 3 partitions on 1 disk then the requirement is for 1 disk

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+    )
+
+    cache_device.create_partitions([Size(500, Unit.MebiByte)])
+    core_device.create_partitions([Size(1, Unit.GigaByte), Size(1, Unit.GigaByte), Size(1, Unit.GigaByte)])
```
Change to Gibi (otherwise will not be aligned)

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+
+    Udev.disable()
+
+    TestProperties.LOGGER.info(f"Staring cache")
```
Starting

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+        allocation=True,
+        rule=f"unclassified",
+        ioclass_config_path=ioclass_config_path,
+    )
```
maybe just:
`ioclass_config.create_ioclass_config(
        add_default_rule=True, ioclass_config_path=ioclass_config_path
    )`
instead of creating config without rule and adding default one (with ioclass_id=1 for some reason)

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+            .oflag("direct")
+        )
+        # Since every IO has the same size, every stat should be increased with the same step.
+        # So there is no need to keep value of every stat in separte variable
```
separate

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+            (CacheMode.WA, read_wa_zero_stats), (CacheMode.WO, read_wo_zero_stats)],
+)
+@pytest.mark.parametrize(
+    "prepare_and_cleanup", [{"core_count": 3, "cache_count": 1}], indirect=True
```
"core_count": 1

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+    iterations = 10
+    dd_size = Size(4, Unit.KibiByte)
+    dd_count = 10
+    dd_skip = 0
```
dd_skip not used

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+
+    flush_cache(cache_id)
+
+    # Check stats for cache in wb after performing write operation
```
wb?

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+
+    flush_cache(cache_id)
+
+    # Check stats for cache in wb after performing read operation
```
wb?

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,261 @@
+            .iflag("direct")
+        )
+        # Since every IO has the same size, every stat should be increased with the same step.
+        # So there is no need to keep value of every stat in separte variable
```
separate

Deixx commented on test_utils/filesystem/file.py:
```diff
@@ -38,6 +40,14 @@ def create_file(path):
         output = fs_utils.ls(f"{path}")
         return fs_utils.parse_ls_output(output)[0]
 
+    def random_padding(self, size: Size):
```
looks like there is no randomness here ;)

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,334 @@
+from api.cas.casadm import StatsFilter
+from api.cas import casadm
+from api.cas import ioclass_config
+from api.cas import casadm_parser
```
casadm_parser not used

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,334 @@
+                    assert value.get_value(Unit.Blocks4096) == 0, (
+                        f"{key} has invalid value of \n"
+                        f"core id {core.core_id}, i: {i}, dd_size: "
+                        f'{dd_size.get_value(Unit.Blocks4096)}\n"
```
string starts with ' ends with "

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,334 @@
+                    assert assumed_value == value.get_value(Unit.Blocks4096), (
+                        f"{key} has invalid value of {value.get_value(Unit.Blocks4096)}\n"
+                        f"core id {core.core_id}, i: {i}, dd_size: "
+                        f'{dd_size.get_value(Unit.Blocks4096)}\n"
```
string starts with ' ends with "

Deixx commented on test_package/stats/test_block_stats.py:
```diff
@@ -0,0 +1,334 @@
+from test_package.test_properties import TestProperties
+from storage_devices.disk import DiskType
+from test_utils.size import Size, Unit
+from test_utils.os_utils import sync, Udev
```
sync not used

Deixx commented on test_package/stats/test_ioclass_stats.py:
```diff
@@ -0,0 +1,182 @@
+    "prepare_and_cleanup", [{"core_count": 1, "cache_count": 1}], indirect=True
+)
+def test_ioclass_stats_set(prepare_and_cleanup):
+    """Try to retvie stats for all set ioclasses"""
```
retrieve

Deixx commented on test_package/stats/test_ioclass_stats.py:
```diff
@@ -0,0 +1,182 @@
+    )
+    for i in range(min_ioclass_id, max_ioclass_id):
+        ioclass_config.add_ioclass(
+            ioclass_id=(i),
```
redundant parenthesis

Deixx commented on test_package/stats/test_ioclass_stats.py:
```diff
@@ -0,0 +1,182 @@
+    ioclass_id_list.append(0)
+    TestProperties.LOGGER.info("Copying files to mounted core and stats check")
+    for f in files_list:
+        # To prevent stats pollution by fielsystem requests, umount core device
```
filesystem

Deixx commented on test_package/stats/test_ioclass_stats.py:
```diff
@@ -0,0 +1,182 @@
+@pytest.mark.parametrize(
+    "prepare_and_cleanup", [{"core_count": 1, "cache_count": 1}], indirect=True
+)
+def test_ioclass_stats_sum(prepare_and_cleanup):
```
Error during execution:
                    cache_stats[stat_name] -= ioclass_stats[stat_name]
E                   KeyError: 'reads from core(s)'

mmichal10 replied:
Ioclass block stats are extended with new entries, however patch is not merged into master yet - thats why this error occurred.

Deixx commented on test_package/stats/test_ioclass_stats.py:
```diff
@@ -0,0 +1,182 @@
+    "prepare_and_cleanup", [{"core_count": 1, "cache_count": 1}], indirect=True
+)
+def test_ioclass_stats_sum(prepare_and_cleanup):
+    """Check if stats for all set ioclasses sums up to cache stats"""
```
sum up

Deixx committed:
Fix log gathering

For DUTs without IP defined use host when creating log dir
(IP set to None caused an exception)
Use paramiko method for copying log files
(works when running tests from Windows)

Signed-off-by: Daniel Madej <daniel.madej@huawei.com>
